pageserver/src/tenant/timeline.rs

```diff
@@ -4218,3 +4224,10 @@ fn rename_to_backup(path: &Path) -> anyhow::Result<()> {
 
     bail!("couldn't find an unused backup number for {:?}", path)
 }
+
+pub fn debug_assert_current_span_has_tenant_and_timeline_id() {
+    let current = tracing::span::Span::current();
+    // would be nice to assert the values, but, that's not possible
+    debug_assert!(current.has_field("tenant_id"));
+    debug_assert!(current.has_field("timeline_id"));
+}
```

Comment:
I remember looking that to do the whole hierarchy inspection is a bit non-trivial, but this is a pragmatic first step!

Reply:
Yeah, it's not as straight-forward as I had hoped.

I went down the rabbit hole, pushing soon

Reply:
Pushed a change that implements proper hierarchy inspection.